op-deployer/pkg/deployer/manage/migrate.go

```diff
@@ -0,0 +1,179 @@
+	oplog "github.com/ethereum-optimism/optimism/op-service/log"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/lmittmann/w3"
+	"github.com/urfave/cli/v2"
+)
+
+type InteropMigrationInput struct {
+	Prank common.Address `json:"prank"`
+	Opcm  common.Address `json:"opcm"`
+
+	UsePermissionlessGame          bool           `json:"usePermissionlessGame"`
+	StartingAnchorRoot             common.Hash    `json:"startingAnchorRoot"`
+	StartingAnchorL2SequenceNumber *big.Int       `json:"startingAnchorL2SequenceNumber"`
+	Proposer                       common.Address `json:"proposer"`
+	Challenger                     common.Address `json:"challenger"`
+	MaxGameDepth                   uint64         `json:"maxGameDepth"`
+	SplitDepth                     uint64         `json:"splitDepth"`
+	InitBond                       *big.Int       `json:"initBond"`
+	ClockExtension                 uint64         `json:"clockExtension"`
+	MaxClockDuration               uint64         `json:"maxClockDuration"`
+
+	EncodedChainConfigs []OPChainConfig `evm:"-" json:"chainConfigs"`
+}
+
+func (u *InteropMigrationInput) OpChainConfigs() ([]byte, error) {
+	data, err := opChainConfigEncoder.EncodeArgs(u.EncodedChainConfigs)
+	if err != nil {
+		return nil, fmt.Errorf("failed to encode chain configs: %w", err)
+	}
+	return data[4:], nil
+}
+
+type OPChainConfig struct {
+	SystemConfigProxy common.Address `json:"systemConfigProxy"`
+	ProxyAdmin        common.Address `json:"proxyAdmin"`
+	AbsolutePrestate  common.Hash    `json:"absolutePrestate"`
+}
+
+type InteropMigrationOutput struct {
+	DisputeGameFactory common.Address `json:"disputeGameFactory"`
+}
+
+func (output *InteropMigrationOutput) CheckOutput(input common.Address) error {
+	return nil
+}
+
+var opChainConfigEncoder = w3.MustNewFunc("dummy((address systemConfigProxy,address proxyAdmin,bytes32 absolutePrestate)[])", "")
+
+type InteropMigration struct {
+	Run func(input common.Address)
+}
+
+func Migrate(host *script.Host, input InteropMigrationInput) (InteropMigrationOutput, error) {
+	return opcm.RunScriptSingle[InteropMigrationInput, InteropMigrationOutput](host, input, "InteropMigration.s.sol", "InteropMigration")
+}
+
+func MigrateCLI(cliCtx *cli.Context) error {
+	logCfg := oplog.ReadCLIConfig(cliCtx)
+	lgr := oplog.NewLogger(oplog.AppOut(cliCtx), logCfg)
+	oplog.SetGlobalLogHandler(lgr.Handler())
+
+	ctx, cancel := context.WithCancel(cliCtx.Context)
+	defer cancel()
+
+	l1RPCUrl := cliCtx.String(deployer.L1RPCURLFlag.Name)
+	if l1RPCUrl == "" {
+		return fmt.Errorf("missing required flag: %s", deployer.L1RPCURLFlag.Name)
+	}
+
+	privateKey := cliCtx.String(deployer.PrivateKeyFlag.Name)
+	privateKeyECDSA, err := crypto.HexToECDSA(strings.TrimPrefix(privateKey, "0x"))
+	if err != nil {
+		return fmt.Errorf("failed to parse private key: %w", err)
+	}
```

Comment:
Should we make this flag optional? I'm wondering if there are situations where we would only want to print the calldata without sending the tx. With other opcm txs we expect the tx to be sent via a gnosis safe, so we just print the calldata and force the user to deal with sending the tx separately